packages/regl-worldview/src/utils/aggregate.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+
+//  Copyright (c) 2018-present, GM Cruise LLC
+//
+//  This source code is licensed under the Apache License, Version 2.0,
+//  found in the LICENSE file in the root directory of this source tree.
+//  You may not use this file except in compliance with the License.
+
+// Takes an array of [value, key] and aggregates across the keys. Results in a Map of [key, values[]], in order of the
+// keys as seen in the array.
+export default function aggregate<T, K>(array: Array<[T, K]>): Map<K, T[]> {
+  const aggregationMap = new Map<K, T[]>();
+  array.forEach(([item, key]) => {
+    const existingItems = aggregationMap.get(key) || [];
+    existingItems.push(item);
+    aggregationMap.set(key, existingItems);
+  });
+  // Since later `set` calls overwrite the earlier ones in ordering, we have to use a new map to ensure that the order
```

Comment:
Seems like you don't need to call `set()` every time, only if it's the first time you're adding a key, so you could avoid this second map.